monai/losses/dice.py

```diff
@@ -23,63 +23,82 @@
-    calculated. The `smooth` parameter is a value added to the intersection and union components of the inter-over-union
-    calculation to smooth results and prevent divide-by-0, this value should be small. The `include_background` class
-    attribute can be set to False for an instance of DiceLoss to exclude the first category (channel index 0) which is
-    by convention assumed to be background. If the non-background segmentations are small compared to the total image
-    size they can get overwhelmed by the signal from the background so excluding it in such cases helps convergence.
+    Compute average Dice loss between two tensors. It can support both multi-classes and multi-labels tasks.
     """
 
-    def __init__(self, include_background=True, do_sigmoid=False, do_softmax=False):
+    def __init__(
+        self,
+        include_background=True,
+        to_onehot_y=False,
+        do_sigmoid=False,
+        do_softmax=False,
+        squared_pred=False,
+        jaccard=False
+    ):
         """
         Args:
             include_background (bool): If False channel index 0 (background category) is excluded from the calculation.
+            to_onehot_y (bool): whether to convert `y` into the one-hot format. Defaults to False.
             do_sigmoid (bool): If True, apply a sigmoid function to the prediction.
             do_softmax (bool): If True, apply a softmax function to the prediction.
+            squared_pred (bool): use squared versions of targets and predictions in the denominator or not.
+            jaccard (bool): compute Jaccard Index (soft IoU) instead of dice or not.
+
         """
         super().__init__()
         self.include_background = include_background
+        self.to_onehot_y = to_onehot_y
         if do_sigmoid and do_softmax:
             raise ValueError('do_sigmoid=True and do_softmax=Ture are not compatible.')
         self.do_sigmoid = do_sigmoid
         self.do_softmax = do_softmax
+        self.squared_pred = squared_pred
+        self.jaccard = jaccard
 
     def forward(self, pred, ground, smooth=1e-5):
-        if ground.shape[1] != 1:
-            raise ValueError("Ground truth should have only a single channel, shape is " + str(ground.shape))
-
-        psum = pred.float()
-        if self.do_sigmoid:
-            psum = psum.sigmoid()  # use sigmoid activation
-        if pred.shape[1] == 1:
-            if self.do_softmax:
-                raise ValueError('do_softmax is not compatible with single channel prediction.')
-            if not self.include_background:
+        """
+        Args:
+            pred (tensor): the shape should be BNH[WD].
+            ground (tensor): the shape should be BNH[WD].
+            smooth (float): a small constant to avoid nan.
+        """
+        n_pred_ch = pred.shape[1]
+        if self.do_sigmoid is True:
+            pred = torch.sigmoid(pred)
+        if self.do_softmax is True:
```

Comment:
when both do_softmax=True and do_sigmoid =True, it should raise a ValueError

Reply:
checked in __init__().
Thanks.